handlers/spaces.py

```
@@ -0,0 +1,650 @@
+
+                if summary_response.status_code == 200:
+                    summary_data = summary_response.json()
+                    logger.info(f"Summarization response: {summary_data}")
+                    return True, summary_data.get('summary', '✅ Space summarized successfully!')
+                else:
+                    logger.error(f"Failed to summarize space. Status code: {summary_response.status_code}, Response: {summary_response.text}")
+                    return False, f"❌ Failed to summarize space: {summary_response.text}"
+            elif job_status == 'failed':
+                raise PermanentError(f"Job failed: {data.get('job', {}).get('error', 'Unknown error')}")
+            else:
+                raise TransientError("Job still processing") from e
```

Ruff (0.8.2)

Undefined name `e`

(F821)